docs/API.md

```diff
@@ -0,0 +1,176 @@
+* steps - the list of steps for this action, including the status for that step.
+
+#### GET
+Returns the list of actions in the system that have been posted, and are accessible to the current user.
+
+##### Responses
+* 200 OK  
+If the actions can be retrieved.
+
+#### POST
+Creates an action in the system. This will cause some action to start.
+The input body to this post will represent an action object that has at least these fields:
+* name  
+The name of the action to invoke. This is likely to map to the actual DAG Names, but may be mapped for "nicer" values.
+* parameters  
+A dictionary of parameters to use for the trigger invocation. The supported parameters will vary for the action invoked.
+
+The POST will synchrounously create the action (a shell object that represents a DAG invocation), perform any checks to validate the preconditions to run the DAG, and trigger the invocation of the DAG.
+The DAG will run Asynchrounously in airflow.
+##### Responses
+* 201 Created  
+If the action is created successfully, and all preconditions to run the DAG are successful. The response body is the action entity created.  
+* 400 Bad Request  
+If the action name doesn't exist, or the payload is otherwise malformed.  
+* 409 Conflict  
+For any failed validations. The response body is the action entity created, with the failed validations. The DAG will not begin execution in this case.  
+
+
+### /v1.0/actions/{action_id} 
+Each action will be assigned an unique id that can be used to get details for the action, including the execution status.
+
+#### GET
+Returns the action entity for the specified id.
+##### Responses
+* 200 OK 
+
+
+### /v1.0/actions/{action_id}/validationDetails/{validation_id} 
+Allows for drilldown to validation detailed info.
+
+#### GET
+Returns the validation detail by Id for the supplied action Id.
+##### Responses
+* 200 OK 
+
+
+### /v1.0/actions/{action_id}/steps/{step_id}
+Allow for drilldown to step information. The step information includes details of the steps excution, successful or not, and enough to facilitate troubleshooting in as easy a fashion as possible. 
+
+#### GET
+Returns the details for a step by id for the given action by Id.
+##### Responses
+* 200 OK 
+
+
+### /v1.0/actions/{action_id}/{control_verb}
+Allows for issuing DAG controls against an action.
+
+#### POST 
+Trigger a control action against an activity.- this includes: pause, unpause, abandon
+##### Responses
+* 202 Accepted
+
+---
+## <a name="AirflowMonitoringAPI"></a> Airflow Monitoring API
+Airflow has a primary function of scheduling DAGs, as opposed to Shipyard's primary case of triggering DAGs.  Shipyard will need to provide functionality to allow for an operator to monitor and review the scheduled tasks.  This API will allow for accessing Airflow DAGs of either type.
+
+### /v1.0/scheduledDAGs
+The resource that represents DAGs in airflow
```

Comment:
IMO `scheduled` has a connotation as something set to happen at a known point in the future. It appears this endpoint is actually for querying DAGs that have already run in the recent past. Maybe something like `completedJobs` instead indicating this is for querying DAGs that have already run.